syntax = "proto3";

package orderqueue;

message User {
  string name = 1;
  string contact = 2;
}

message CreditCard {
  string number = 1;
  string expirationDate = 2;
  string cvv = 3;
}

message OrderRequest {
  User user = 1;
  CreditCard creditCard = 2;
  string userComment = 3;
  // Include other fields as needed
}

message EnqueueResponse {
  bool success = 1;
  string message = 2;
}

service OrderQueueService {
  rpc EnqueueOrder(OrderRequest) returns (EnqueueResponse);
}
